packages/vite/src/node/cli.ts

```diff
@@ -132,19 +132,32 @@ cli
   .option('-m, --mode <mode>', `[string] set env mode`)
   .option('-w, --watch', `[boolean] rebuilds when modules have changed on disk`)
   .action(async (root: string, options: BuildOptions & GlobalCLIOptions) => {
-    const { build } = await import('./build')
     const buildOptions = cleanOptions(options) as BuildOptions
 
     try {
-      await build({
-        root,
-        base: options.base,
-        mode: options.mode,
-        configFile: options.config,
-        logLevel: options.logLevel,
-        clearScreen: options.clearScreen,
-        build: buildOptions
-      })
+      if (buildOptions.watch) {
+        const { watch } = await import('./build')
```

Comment:
Is it possible to return an `watch` from the function? Sometimes there is a need to start watching files that are not imported directly in the bundle, or are outside the root. The same access to the API would be useful

```js
const watcher = await build({ ... })
watcher.add('new-file');
```